equinix/resource_metal_port_acc_test.go

```diff
@@ -340,3 +378,113 @@ func testAccMetalPortDestroyed(s *terraform.State) error {
 	}
 	return nil
 }
+
+func testAccWaitForPortActive(deviceName, portName string) resource.ImportStateIdFunc {
+	return func(state *terraform.State) (string, error) {
+		rs, ok := state.RootModule().Resources[deviceName]
+		if !ok {
+			return "", fmt.Errorf("Device Not found in the state: %s", deviceName)
+		}
+		if rs.Primary.ID == "" {
+			return "", fmt.Errorf("No Record ID is set")
+		}
+
+		meta := testAccProvider.Meta()
+		rd := new(schema.ResourceData)
+		meta.(*Config).addModuleToMetalUserAgent(rd)
+		client := meta.(*Config).metal
+		device, _, err := client.Devices.Get(rs.Primary.ID, &packngo.GetOptions{Includes: []string{"ports"}})
+		if err != nil {
+			return "", fmt.Errorf("error while fetching device with Id [%s], error: %w", rs.Primary.ID, err)
+		}
+		if device == nil {
+			return "", fmt.Errorf("Not able to find devices with Id [%s]", rs.Primary.ID)
+		}
+		if len(device.NetworkPorts) == 0 {
+			return "", fmt.Errorf("Found no ports for the device with Id [%s]", rs.Primary.ID)
+		}
+
+		for _, port := range device.NetworkPorts {
+			if port.Name == portName {
+				return port.ID, nil
+			}
+		}
+
+		return "", fmt.Errorf("No port with name [%s] found", portName)
+	}
+}
+
+func TestAccMetalPortCreate_hybridBonded_timeout(t *testing.T) {
+	rs := acctest.RandString(10)
+	rInt := acctest.RandInt()
+	deviceName := "equinix_metal_device.test"
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
This test is [failing in CI](https://github.com/equinix/terraform-provider-equinix/actions/runs/6124500596/job/16624681207#step:5:2485), but I'm not sure if it's a persistent failure or tied to something else:
```
=== NAME  TestAccMetalPortCreate_hybridBonded_timeout
    resource_metal_port_acc_test.go:422: Step 3/3 error: Error running destroy: exit status 1
        
        Error: API Error HTTP 422 Cannot delete Virtual Network when port is assigned
```

Reply:
I don't see any 422s in the latest CI run, so I'm going to chalk this up to intermittent issues